tensorboard/webapp/metrics/views/card_renderer/card_view_container.ts

```diff
@@ -88,7 +88,22 @@ export class CardViewContainer {
   }
 
   onPinStateChanged() {
-    this.store.dispatch(actions.cardPinStateToggled({cardId: this.cardId}));
+    this.store
+      .select(selectors.getCardPinnedState, this.cardId)
+      .pipe(
+        take(1),
+        withLatestFrom(this.store.select(selectors.getCanCreateNewPins)),
+        tap(([wasPinned, canCreateNewPins]) => {
```

Comment:
Maybe do what's inside the `tap` in subscribe instead.

Reply:
Done.